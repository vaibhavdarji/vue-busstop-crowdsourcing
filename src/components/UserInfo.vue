<template>
 <v-card>
   <v-form ref="form" @submit.prevent="onNext">
     <v-container grid-list-xl fluid>
       <v-layout row>
         <v-flex xs12>
           <h2>
           You selected amount:  
            <v-icon left>attach_money</v-icon>
            {{userDonation.amount}}
           </h2>
         </v-flex>  
       </v-layout>
       <v-layout row>
         <v-flex xs2 pl-0>
           <v-subheader>Full Name: </v-subheader>
         </v-flex>
         <v-flex xs8>
           <v-text-field 
              label="Full Name"
              :rules="rules.name"
              v-model="form.userName"
              required
            ></v-text-field>
         </v-flex>
       </v-layout>
       <v-layout row>
         <v-flex xs2 pl-0>
           <v-subheader>Email: <small>(optional)</small></v-subheader>
         </v-flex>
         <v-flex xs8>
           <v-text-field 
              v-model="form.userEmail"
              label="Email"
              type="email"
            ></v-text-field>
         </v-flex>
       </v-layout>
      <v-card-actions>
        <custom-button @onClick="onPrevious">
          Previous 
        </custom-button>
        <v-spacer></v-spacer>
        <custom-button type="submit" :disabled="!formIsValid">
          Next 
        </custom-button>
      </v-card-actions>
     </v-container>
   </v-form>
 </v-card> 
</template>

<script>
import { mapGetters, mapActions} from 'vuex';
import CustomButton from '@/components/Button.vue';
import {
  UPDATE_USER_DONATION_DETAIL
} from '@/actions/types';

export default {
  name: 'UserInfo',
  data () {
    const defaultForm = Object.freeze({
      userName: '',
      userEmail: ''
    });
    return {
      form: Object.assign({}, defaultForm),
      rules: {
        name: [
          value => {
            return (value || '').trim().length > 0 || 'Enter your full name';
          }
        ]
      },
      defaultForm
    };
  },

  components: {
    CustomButton
  },

  computed: {
    ...mapGetters([
      'userDonation'
    ]),

    formIsValid () {
      return (this.form.userName);
    },
  },

  methods: {
    ...mapActions({
      'updateUserDonationDetail': UPDATE_USER_DONATION_DETAIL
    }),
    onPrevious () {
      this.$emit('onPreviousStep');
    },

    onNext () {
      this.updateUserDonationDetail(this.form);
      this.$emit('onNextStep');
    }
  }
}
</script>

