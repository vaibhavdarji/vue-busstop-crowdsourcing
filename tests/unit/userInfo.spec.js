import { shallowMount } from "@vue/test-utils";
import UserInfo from '@/components/UserInfo.vue';

import Vue from "vue";
import Vuex from "vuex";
import Vuetify from "vuetify";
import Constants from '@/constants';

Vue.use(Vuex);
Vue.use(Vuetify);

describe("UserInfo.vue", () => {
  let getters, store;
  beforeEach(() => {
    getters = {
      userDonation: () => ({
        ...Constants.USER_INFO
      })
    };
    store = new Vuex.Store({
      getters
    });
  });

  it("should match snapshot", () => {
    const wrapper = shallowMount(UserInfo, {
      store,
      Vue
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});