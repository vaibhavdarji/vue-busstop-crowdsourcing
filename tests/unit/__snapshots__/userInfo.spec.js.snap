// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo.vue should match snapshot 1`] = `
<v-card-stub
  tag="div"
>
  <v-form-stub>
    <v-container-stub
      fluid=""
      grid-list-xl=""
      tag="div"
    >
      <v-layout-stub
        row=""
        tag="div"
      >
        <v-flex-stub
          tag="div"
          xs12=""
        >
          <h2>
            
          You selected amount:  
           
            <v-icon-stub
              left=""
            >
              attach_money
            </v-icon-stub>
            
           0
          
          </h2>
        </v-flex-stub>
      </v-layout-stub>
       
      <v-layout-stub
        row=""
        tag="div"
      >
        <v-flex-stub
          pl-0=""
          tag="div"
          xs2=""
        >
          <v-subheader-stub>
            Full Name: 
          </v-subheader-stub>
        </v-flex-stub>
         
        <v-flex-stub
          tag="div"
          xs8=""
        >
          <v-text-field-stub
            label="Full Name"
            required=""
            rules="value => {
          return (value || '').trim().length > 0 || 'Enter your full name';
        }"
            value=""
          />
        </v-flex-stub>
      </v-layout-stub>
       
      <v-layout-stub
        row=""
        tag="div"
      >
        <v-flex-stub
          pl-0=""
          tag="div"
          xs2=""
        >
          <v-subheader-stub>
            Email: 
            <small>
              (optional)
            </small>
          </v-subheader-stub>
        </v-flex-stub>
         
        <v-flex-stub
          tag="div"
          xs8=""
        >
          <v-text-field-stub
            label="Email"
            type="email"
            value=""
          />
        </v-flex-stub>
      </v-layout-stub>
       
      <v-card-actions-stub>
        <custom-button-stub
          btntype="button"
        >
          
         Previous 
       
        </custom-button-stub>
         
        <v-spacer-stub />
         
        <custom-button-stub
          btntype="button"
          disabled="true"
          type="submit"
        >
          
         Next 
       
        </custom-button-stub>
      </v-card-actions-stub>
    </v-container-stub>
  </v-form-stub>
</v-card-stub>
`;
